### DISCUSSION TITLE
Using custom image loader in next.config.js

### DISCUSSION THREAD

**Discussion by glaskawiec** [vercel/next.js#19225] [Help] [answered]
**Edit:** Answered below https://github.com/vercel/next.js/discussions/19225#discussioncomment-1040247

Hello, I saw there is information in docs about using custom image loader but the question is how to do it?
I tried to pass function in ```next.config.js``` but it's not working.

```
loader: ({ root, src, width, quality }) => {
  // custom loader code
  return root;
}
```

**Comment by jamesmosier** [comment, marked as answer]
From the looks of the type definition, the `loader` value can only be a string and not a function:

https://github.com/vercel/next.js/blob/5562daf7a160e62fd4f78eca257ba957ad64074c/packages/next/next-server/server/image-config.ts#L1-L16

The accepted strings are:

```
'default',
'imgix',
'cloudinary',
'akamai'
```

**Reply by weyert** [reply]
That's a shame

**Reply by glaskawiec** [reply]
@jamesmosier are you planning to add such possibility?

**Reply by avandonkelaar** [reply]
Following, because this is precisely what I'm looking for.

**Comment by sandcastle** [comment]
We run our own image optimisation service, we need to be able to add our own custom loader in order to make use of `next/image`

**Comment by sandcastle** [comment]
Duplicate of #18450

**Comment by leerob** [comment]
This is now possible! https://nextjs.org/docs/basic-features/image-optimization#loader
